internal/controller/monovertexrollout/monovertexrollout_controller.go

```diff
@@ -248,9 +248,6 @@ func (r *MonoVertexRolloutReconciler) reconcile(ctx context.Context, monoVertexR
 			if err != nil {
 				return ctrl.Result{}, fmt.Errorf("error processing existing MonoVertex: %v", err)
 			}
-			if requeueDelay > 0 {
-				return ctrl.Result{RequeueAfter: requeueDelay}, nil
-			}
```

Comment:
there was no need for this to be here. We are checking below on line 276 for `requeueDelay > 0` and requeueing there. Having it here prevented processing monovertex status below and garbage collecting children.